projects/defisaver.js

```diff
@@ -325,8 +363,156 @@ async function tvl(ts, block) {
   }
 }
 
+async function getOptimismTvl() {
+  return getTvlAndBorrowedFromChain("optimism", 10, true);
+}
+
+async function getOptimismBorrowed() {
+  return getTvlAndBorrowedFromChain("optimism", 10, false);
+}
+
+async function getArbitrumTvl() {
+  return getTvlAndBorrowedFromChain("arbitrum", 42161, true);
+}
+
+async function getArbitrumBorrowed() {
+  return getTvlAndBorrowedFromChain("arbitrum", 42161, false);
+}
+
+async function getTvlAndBorrowedFromChain(chain, chainId, returnTvl) {
+  const balances = {};
+  const debt = {};
+  const prices = (await utils.getPrices(keys)).data;
+  await getAaveV3Data(returnTvl);
+  if (returnTvl) {
+    return balances;
+  } else {
+    return debt;
+  }
+
+  async function getAaveV3Data(returnTvl) {
+    const defaultMarket = networkMapping[chainId].defaultAaveMarketAddress;
+    const startBlock = networkMapping[chainId].startBlock;
+    const provider = getProvider(chain);
+    const endBlock = await provider.getBlockNumber();
+    const topic = "RecipeEvent(address,string)";
+    const topic1 = "ActionDirectEvent(address,string,bytes)";
+    const loggerContract = getAddressByChain(DefisaverLogger, chainId);
+    const aaveV3ViewAddress = getAddressByChain(AaveV3View, chainId);
+    // get proxy contracts by looking through the events : RecipeEvent(address,string)
+    const recipeEvents = (
+      await sdk.api.util.getLogs({
```

Comment:
@kfukue thanks for the PR, fetching logs is bit of an issue on any chain that is not Ethereum. These queries become slower over time and stop working eventually. So, I reject any PRs with this, sorry.

Can you check if they have some sort of graph api for fetching this information?

Reply:
@g1nt0ki No worries. Yea that makes sense. Ok, I'll reach out to the defisaver team to see if there's a graph api or an alternative way to get these proxy contract addresses but if there isn't I'll close the PR until we have a better way to get the proxy contract addresses. Thanks for your feedback!

Reply:
@g1nt0ki I'll close this for now. (not much response from their team)

Reply:
@kfukue yeah, sometimes it is quite frustrating.